lib/ansible/modules/cloud/azure/azure_rm_loadbalancer.py

```diff
@@ -746,14 +862,15 @@ def exec_module(self, **kwargs):
                 backend_address_pools=backend_address_pools_param,
                 probes=probes_param,
                 inbound_nat_pools=inbound_nat_pools_param,
-                load_balancing_rules=load_balancing_rules_param
+                load_balancing_rules=load_balancing_rules_param,
+                inbound_nat_rules=inbound_nat_rules_param
             )
 
             if load_balancer:
+                self.new_load_balancer = self.object_assign(self.new_load_balancer, load_balancer)
+                load_balancer_dict = load_balancer.as_dict()
                 new_dict = self.new_load_balancer.as_dict()
-                if (self.location != load_balancer['location'] or
-                        self.sku != load_balancer['sku']['name'] or
-                        not default_compare(new_dict, load_balancer, '')):
+                if not default_compare(new_dict, load_balancer_dict, ''):
```

Comment:
use set protocol to udp at first run, then not set the value at second run, default will be tcp, compare fail, actually user don't want to change the value

Reply:
Here I keep consistent with all other protocol in this module. I can remove all the default value for it anyway.

Reply:
if parameter is None in the second run, it won't be included in comparison. so this scenario is ok.